virtual_rainforest/models/plants/plants_model.py

```diff
@@ -0,0 +1,147 @@
+"""The :mod:`~virtual_rainforest.models.plants.plants_model` module creates
+:class:`~virtual_rainforest.models.plants.plants_model.PlantsModel` class as a child of
+the :class:`~virtual_rainforest.core.base_model.BaseModel` class.
+"""  # noqa: D205, D415
+
+from __future__ import annotations
+
+from typing import Any
+
+# import numpy as np
+# from numpy.typing import NDArray
+from pint import Quantity
+
+from virtual_rainforest.core.base_model import BaseModel
+from virtual_rainforest.core.data import Data
+
+# from virtual_rainforest.core.config import Config
+# from virtual_rainforest.core.exceptions import InitialisationError
+from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.core.utils import check_valid_constant_names
+from virtual_rainforest.models.plants.community import PlantCommunities
+from virtual_rainforest.models.plants.constants import PlantsConsts
+from virtual_rainforest.models.plants.functional_types import PlantFunctionalTypes
+
+# from xarray import DataArray
+
+
+class PlantsModel(BaseModel):
+    """A class defining the plants model.
+
+    This is currently a basic placeholder to define the main interfaces between the
+    plants model and other models.
+
+    Args:
+        data: The data object to be used in the model.
+        update_interval: Time to wait between updates of the model state.
+        plant_functional_types: A set of plant functional types to be used in the model.
+        constants: Set of constants for the plants model.
+    """
+
+    model_name = "plants"
+    """An internal name used to register the model and schema"""
+    lower_bound_on_time_scale = "1 day"
+    """Shortest time scale that plants model can sensibly capture."""
+    upper_bound_on_time_scale = "1 year"
+    """Longest time scale that plants model can sensibly capture."""
+    required_init_vars = (
+        ("plant_cohorts_n", tuple()),
+        ("plant_cohorts_pft", tuple()),
+        ("plant_cohorts_cell_id", tuple()),
+        ("plant_cohorts_dbh", tuple()),
+    )
+    """Required initialisation variables for the plants model.
+
+    This is the set of variables and their core axes that are required in the data
+    object to create a PlantsModel instance."""
+
+    # TODO - think about a shared "plant cohort" core axis that defines these, but the
+    #        issue here is that the length of this is variable.
+
+    vars_updated = (
+        "leaf_area_index",  # NOTE - LAI is integrated into the full layer roles
+        "layer_heights",  # NOTE - includes soil, canopy and above canopy heights
+        "herbivory",
+    )
```

Comment:
At some point, you will need to add `evapotranspiration` or a derivative here, maybe at a later stage when you actually implement the p-model